### PR TITLE
Replace Next.js with FastAPI, Vite, and react-router in a SPA

### PR DESCRIPTION
**Pull request by alexmojaki**
- FastAPI handles listing and receiving trace files and serving the built SPA.
- Vite is a build tool, basically a more modern alternative to create-react-app that uses rollup instead of webpack and is much faster.
- react-router handles client-side routing to replace the Next router. Instead of pages managed by Next and sometimes rendered on the server, it just chooses which react component to render based on the URL. This will also make it easier to use hash-based routing if we want.

This PR also removes node stuff from the docker workflow because that was a pain to develop with and we want to start moving away from docker. It doesn't finish that process properly though and doesn't update the README to match. I'm thinking of making a new base branch that will contain the broader refactoring until it's complete, with multiple PRs merging into that branch for easier review. This will be the first, followed by some version of #102. The broader goals include moving away from poetry, making a package suitable for PyPI, and maybe moving some of the more cumbersome optional stuff like pytorch into a separate repo. @lslunis how does that sound?

### CODE REVIEW

**Comment by alexmojaki** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by lslunis** [comment]
new base branch sgtm

File: ice/routes/traces.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+
+@router.get("/")
+async def list_traces():
+    # e.g. if trace_dir contains files trace1.jsonl, trace2.jsonl, other.txt,
+    # return ["trace1", "trace2"]
+    return [trace.stem for trace in trace_dir.glob("*.jsonl")]
+
+
+@router.get("/{trace_id}.jsonl")
```
[the remark this answers is not in the document]
Comment (alexmojaki) [reply ice/routes/traces.py:19]: Added a docstring to explain.

File: Dockerfile [hunk 2]
```diff
@@ -42,10 +42,6 @@ ENV PATH="${VIRTUAL_ENV}/bin:${PATH}"
 
 RUN python -c "import nltk; nltk.download('punkt')"
 
-COPY ui/package.json ui/package-lock.json ui/
-COPY ui/patches/*.patch ui/patches/
-RUN npm --prefix ui ci
-
 COPY . .
 
-CMD ["concurrently", "uvicorn ice.routes.app:app --host 0.0.0.0 --port 8935 --reload", "npm --prefix ui run dev"]
```
[the remark this answers is not in the document]
Comment (alexmojaki) [reply Dockerfile:51]: I think for the sake of scope it'd be better to handle that later. I only wanted to get docker out of my way at this stage.

**Review by lslunis — APPROVED** [review]

**Pull request merged**